pkg/plugin/connector/builtin/stream.go

```diff
@@ -0,0 +1,181 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/conduitio/conduit-connector-protocol/pconnector"
+)
+
+type InMemoryDestinationRunStream struct {
+	stream *inMemoryStream[pconnector.DestinationRunRequest, pconnector.DestinationRunResponse]
+}
+
+func (s *InMemoryDestinationRunStream) Init(ctx context.Context) {
```

Comment:
Small nit. What do you think about naming these receivers? `im` instead?

Reply:
I don't feel strongly one way or the other. I can give you the reasoning why I used `s` - it's a stream 😉